contracts/NonFungibleToken.cdc

```diff
@@ -194,7 +197,16 @@ access(all) contract interface NonFungibleToken: ViewResolver {
 
         /// Gets the amount of NFTs stored in the collection
         /// @return An integer indicating the size of the collection
-        access(all) view fun getLength(): Int
+        access(all) view fun getLength(): Int {
+            return self.ownedNFTs.length
+        }
+
+        /// Returns an iterator that allows callers to iterate
+        /// through the list of owned NFT IDs in a collection
+        /// without having to load the entire list first
+        access(all) fun forEachID(_ f: fun (UInt64): Bool): Void {
+            self.ownedNFTs.forEachKey(f)
```

Comment:
If I define a function that removes the NFT from the mapping and either destroys or transfers it, would that be possible? Maybe we add a post condition on length to the method

Reply:
@sisyphusSmiling i dont fully understand, by post condition do you mean an assert inside of the `forEachID` method?

Reply:
That all makes sense, thanks for clarifying. I was initially spooked to see a non-view method running arbitrary code on `self.ownedNFTs` within interface scope, but I can see how a given method would still need entitled access on the underlying collection or `ownedNFTs` mapping. The post-condition was misguided, so no worries there.